lib/features/pay/pages/pay_page.dart

```diff
@@ -68,16 +74,20 @@ class PayPageState extends State<PayPage> {
                 _buildNumberPad(),
                 ElevatedButton(
                   onPressed: () {
-                    Navigator.pushNamed(context, '/pay_who', arguments: {'amount': amount, 'isTopUp': isTopUp}); 
-
+                    _logger.info(
+                        'Proceeding with payment: amount=$amount, isTopUp=${widget.isTopUp}');
+                    context.go('/pay_who',
```

Comment:
I think this needs to be `push()`, to not replace the entire stack of pages with this one. Go over all occurances of this - `go(...)` should only be used to replace the entire stack, like when you logout, or maybe finish the payment and need to go to the home page. Here, if user on the next page decides to go back, he should be able to.